core/src/main/java/com/datastrato/gravitino/connector/BaseCatalog.java

```diff
@@ -169,12 +180,60 @@ public CatalogOperations ops() {
     return ops;
   }
 
+  public AuthorizationPlugin getAuthorizationPlugin() {
+    if (authorization == null) {
+      synchronized (this) {
+        if (authorization == null) {
+          authorization = createAuthorizationPluginInstance();
+        }
+      }
+    }
+    return authorization.plugin();
+  }
+
+  private BaseAuthorization<?> createAuthorizationPluginInstance() {
+    AuthorizationPluginType pluginType =
+        (AuthorizationPluginType)
+            catalogPropertiesMetadata().getOrDefault(conf, AUTHORIZATION_PLUGIN);
+    if (pluginType == null) {
```

Comment:
You have brought in two concepts, one is "provider", another is "type", so this confuses people a lot, I think we should only use one for all, like "provider".

Reply:
Fixed.